### PR TITLE
Support use of MarkDups BAMs in oncoanalyser_wgts_existing_both_step

### PR DESCRIPTION
**Pull request by scwatts**
### Background

- this change accommodates the new MarkDups stage in oncoanalyser dev [20240711; 1.0.0 release candidate]
- the MarkDups stage takes the input DRAGEN WGS BAMs and generates processed BAMs that are used for analysis
- it is therefore expected that when running with existing inputs the MarkDups WGS BAMs are provided to oncoanalyser

### Changes

A high-level summary of changes is provided in the below figure (shown in red).

image

_Detailed changes_

- oncoanalyser_wgts_existing_both_step orchestrator now discovers MarkDups WGS BAMs from existing WGS directory
- discovered MarkDups WGS BAMs replace DRAGEN WGS BAMs as payload input to the corresponding Lambda function
- an error will be raised when no MarkDups WGS BAMs are found
  -  this intentionally requires users to run oncoanalyser WGS to generate the appropriate MarkDups WGS BAMs
  - and doing so will have oncoanalyser WGTS (existing both) automatically trigger assuming WTS outputs are present
  - this gives backwards compatibility and guards against errors caused where patients have an incomplete complement of WGS/WTS data then receive their respective WGS or WTS sample following this change
- new tests have been implemented for the new functionality

### Other notes

- some consider needs to be made so that MarkDups BAMs are subject to our internal lifecycle policies for this filetype

### CODE REVIEW

**Comment by victorskl** [comment]
I will review very soon, Steve. ETA mid next week. 

Let me push current changes through to PROD and release the v2.2.2 first. We are almost there.

**Comment by scwatts** [comment]
Something else to discuss when you get back to this @victorskl is whether/where we should document that MarkDups WGS BAMs will now always be expected for oncoanalyser jobs that run from existing WGS results.

**Comment by victorskl** [comment]
As discussed, I will review it on next Monday first thing. Mark dups support is in next Portal high priority list. 

Also. As Portal [release-v2.2.2](https://github.com/umccr/data-portal-apis/releases/tag/2.2.2) just made recently, we would want current release to pass through at least (this weekend) 1 round of sequencing run (ditto [code freeze window](https://github.com/umccr/data-portal-apis/blob/dev/docs/PORTAL_RELEASE.md)) before merging feature branch. This preserves for any urgent hot fix, if any.

File: data_processors/pipeline/orchestration/tests/test_oncoanalyser_wgts_existing_both_step.py [hunk 1]
```diff
@@ -288,3 +322,43 @@ def test_get_existing_wts_dir_raises_error(self):
...
+        _ = OncoanalyserWgsS3ObjectOutputFactory()
+
+        mock_existing_wgs_dir = oncoanalyser_wgts_existing_both_step.get_existing_wgs_dir(mock_wgs_wfl)
+
+        with self.assertRaises(ValueError) as cm:
+            _ = oncoanalyser_wgts_existing_both_step.get_existing_wgs_markdups_bam(mock_existing_wgs_dir, TestConstant.sample_id.value)
+        e = cm.exception
+
+        logger.exception(f"THIS ERROR EXCEPTION IS INTENTIONAL FOR TEST. NOT ACTUAL ERROR. \n{str(e)}")
+        self.assertIn("No MarkDups BAM found", str(e))
```
Comment (victorskl) [inline data_processors/pipeline/orchestration/tests/test_oncoanalyser_wgts_existing_both_step.py:364]: Nice! Thanks for unit testing.

**Review by victorskl — APPROVED** [review]
Looking good!

**Comment by victorskl** [comment]
We will let current CUP samples processing done with oncoanalyser in dev before merging PR.

Around +/- here: https://umccr.slack.com/archives/C7QC9N8G4/p1721691075739399

**Comment by scwatts** [comment]
See Slack for tests and related discussion

- https://umccr.slack.com/archives/C7QC9N8G4/p1722133586763919
- https://umccr.slack.com/archives/C7QC9N8G4/p1722210764879359
- https://umccr.slack.com/archives/C7QC9N8G4/p1722219035333969

**Pull request merged**